Code/Framework/AzFramework/AzFramework/DocumentPropertyEditor/AggregateAdapter.cpp

```diff
@@ -251,11 +251,37 @@ namespace AZ::DocumentPropertyEditor
                 else
                 {
                     // TODO: handle row being replaced
+                    AZ_Assert(0, "AggregateAdapter: row replace operation is not supported yet!");
                 }
             }
             else if (operationIterator->GetType() == AZ::Dom::PatchOperation::Type::Add)
             {
-                // TODO: handle new nodes being added
+                auto destinationPath = operationIterator->GetDestinationPath();
+                auto lastPathEntry = destinationPath.Back();
+                destinationPath.Pop();
+                auto parentNode = GetNodeAtAdapterPath(adapterIndex, destinationPath);
+                AZ_Assert(parentNode, "can't find node to add to!");
+                if (parentNode)
+                {
+                    if (IsRow(operationIterator->GetValue()))
+                    {
+                        // adding a full row
+                        AZ_Assert(lastPathEntry.IsIndex(), "new addition is a row, it must be addressed by index!");
+                        auto childIndex = lastPathEntry.GetIndex();
+                        auto* aggregateNode = AddChildRow(adapterIndex, parentNode, operationIterator->GetValue(), childIndex);
+                        if (aggregateNode && aggregateNode->EntryCount() == m_adapters.size())
```

Comment:
question: Just to understand this better, in what cases will the entry count be not equal to the adapters size for adding a child row?